spec/development/workload/PageIterator.ts

```diff
@@ -0,0 +1,61 @@
+import { assert } from "chai";
+import { Event } from "microsoft-graph";
+
+import { PageIterator, PageIteratorCallback } from "../../../src/tasks/PageIterator";
+import { getClient } from "../test-helper";
+
+const client = getClient();
+describe("PageIterator", function() {
+	describe("sameHeadersPassedWithPageIterator", () => {
+		const pstHeader = { Prefer: 'outlook.timezone= "pacific standard time"' };
+		const utc = "UTC";
+		const pst = "Pacific Standard Time";
+
+		it("verify same headers", async () => {
+			const response = await client
+				.api(`/me/events?$top=2`)
+				.headers(pstHeader)
+				.select("id,start,end")
+				.get();
+
+			const callback: PageIteratorCallback = (eventResponse) => {
+				const event = eventResponse as Event;
+				assert.equal(event.start.timeZone, pst);
+				return true;
+			};
+			var requestOptions = { headers: pstHeader };
+			if (response["@odata.nextLink"]) {
+				const pageIterator = new PageIterator(client, response, callback, requestOptions);
+				await pageIterator.iterate();
+				assert.isTrue(pageIterator.isComplete());
+			}
+		});
+
+		it("differentHeadersPassedWithPageIterator", async () => {
```

Comment:
not for this PR but the call pattern seems weird to me.  The fact that I can have different headers and middleware for the first call versus the rest of the iteration.  What I'd expect is just to specify the request and that I wish to iterate it.  The headers and middleware all being the same.

Reply:
https://github.com/microsoftgraph/msgraph-sdk-design/issues/37